wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java

```diff
@@ -109,16 +107,22 @@ public ProjectManager(VirtualFileSystemProvider vfsProvider,
 
     @PostConstruct
     void initWatcher() throws IOException {
-        FileWatcherNotificationListener defaultListener = new FileWatcherNotificationListener(VirtualFileFilter.ACCEPT_ALL) {
-            @Override
-            public void onFileWatcherEvent(VirtualFile virtualFile, FileWatcherEventType eventType) {
-                LOG.debug("FS event detected: " + eventType + " " + virtualFile.getPath().toString() + " " + virtualFile.isFile());
-                eventService.publish(DtoFactory.newDto(VfsWatchEvent.class)
-                                               .withPath(virtualFile.getPath().toString())
-                                               .withFile(virtualFile.isFile())
-                                               .withType(eventType));
-            }
-        };
+        FileWatcherNotificationListener defaultListener =
+                new FileWatcherNotificationListener(file -> !(file.getPath().toString().contains(".codenvy")
+                                                              || file.getPath().toString().contains(".#"))) {
```

Comment:
AFAIK git performs a lot of file operation in .git folder. We had hard time with git on GlusterFS because of that.
So I would consider exclusion of .git folder from tracking (except HEAD file or other files we may need to track).
Also we can improve that by binding filters to track/untrack some files/paths.